TourMate/TourMate/Models/DomainModels/Plan.swift

```diff
@@ -10,10 +10,10 @@ import Foundation
 protocol Plan {
     var id: String { get }
     var tripId: String { get }
-    var planType: PlanType { get set }
+    var planType: PlanType { get }
     var name: String { get set }
     var startDate: Date { get set }
-    var endDate: Date? { get set }
+    var endDate: Date { get set }
```

Comment:
Any reason why this was changed? We made it optionally as `Restaurant` does not have an `endDate`

Reply:
Oh I was actually following the data model where the endDate is present:

```
protocol FirebaseAdaptedPlan: FirebaseAdaptedData {
    var tripId: String { get }
    var name: String { get }
    var planType: FirebasePlanType { get }
    var startDate: Date { get }
    var endDate: Date { get }
```

Reply:
Why does it not have an endDate? I thought that we will be handling startDate and endDate checks that they fit into a trip down the line